### ISSUE TITLE
[java] ImmutableField: False positive with lombok

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#410]
See comment https://sourceforge.net/p/pmd/bugs/1494/#041f

**Rule Set:**
Design / ImmutableField

**Description:**

with PMD 5.7, I get following:

Private field 'id' could be made final; it is only initialized in the declaration or constructor.

The id could be later set using Setter method.

**Code Sample demonstrating the issue:**

```
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Foo {
    private String id;
    public Foo(String id) {
        this.id = id;
    }
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
